flagd/pkg/service/flag-evaluation/ofrep/handler.go

```diff
@@ -0,0 +1,115 @@
+package ofrep
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/open-feature/flagd/core/pkg/evaluator"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/service/ofrep"
+	"github.com/rs/xid"
+)
+
+const (
+	key              = "key"
+	singleEvaluation = "/ofrep/v1/evaluate/flags/{key}"
+	bulkEvaluation   = "/ofrep/v1/evaluate/{path:flags\\/|flags}"
+)
+
+type handler struct {
+	Logger    *logger.Logger
+	evaluator evaluator.IEvaluator
+}
+
+func NewHandler(logger *logger.Logger, evaluator evaluator.IEvaluator) http.Handler {
+	h := handler{
+		logger,
+		evaluator,
+	}
+
+	router := mux.NewRouter()
+	router.HandleFunc(singleEvaluation, h.HandleFlagEvaluation).Methods("POST")
+	router.HandleFunc(bulkEvaluation, h.HandleBulkEvaluation).Methods("POST")
+	return router
+}
+
+func (h *handler) HandleFlagEvaluation(w http.ResponseWriter, r *http.Request) {
+	requestID := xid.New().String()
+
+	// obtain flag key
+	vars := mux.Vars(r)
+	flagKey := vars[key]
+	if vars == nil || strings.TrimSpace(flagKey) == "" {
+		h.writeJSONToResponse(
+			http.StatusInternalServerError,
```

Comment:
I think this error is a client error and should be HTTP 400 - Bad Request

Reply:
I thought about using 4xx error code, however, this can only happen if our mux configuration or path configurations are incorrect. Otherwise, requests to `ofrep/v1/evaluate/flags/` (missing flag in path) is considered as a bulk evaluation request. This is why I use 500 status code.

Reply:
good point, I was thinking of having a key value equal to `%20`, which is trimmed away 🤔 
Maybe we need to split:
- `vars = nil` -> 500
- empty key -> 400

Reply:
@thisthat this is good point. I update to allow `empty key` as there's no rule or parser that prevents flagd from having a flag key with empty string (ex: `  ` is valid ). And updated vars= nil to be a 500